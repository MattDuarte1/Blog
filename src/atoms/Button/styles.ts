import styled, { css } from 'styled-components';

export const Container = styled.button`
  ${({ theme, disabled }) => css`
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    padding: 20px 23px 19px;
    gap: 10px;
    border: none;
    cursor: ${disabled ? 'not-allowed' : 'pointer'};

    min-width: 145px;
    height: 67px;
    font-weight: 700;
    color: #ffff;

    background: ${theme.colors.blue};
    border-radius: 8px;
    transition: filter 0.4s ease;

    filter: ${disabled && 'brightness(0.7)'};

    &:hover {
      filter: ${!disabled && 'brightness(1.1)'};
    }
  `}
`;
