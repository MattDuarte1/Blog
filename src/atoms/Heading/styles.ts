import styled, { css, DefaultTheme } from 'styled-components';

interface ContainerProps {
  size?:
    | 'xxsmall'
    | 'xsmall'
    | 'small'
    | 'medium'
    | 'large'
    | 'xlarge'
    | 'xxlarge'
    | 'huge'
    | 'xhuge';
  color:
    | 'white'
    | 'lightGrey1'
    | 'lightGrey2'
    | 'grey'
    | 'darkGrey'
    | 'black'
    | 'blue';
  uppercase: boolean;
}
const MediaFont = {
  large: (theme: DefaultTheme) => css`
    @media ${theme.media.lteSmall} {
      font-size: ${theme.font.sizes.medium};
    }
  `,
  xxlarge: (theme: DefaultTheme) => css`
    @media ${theme.media.lteMedium} {
      font-size: ${theme.font.sizes.xlarge};
    }
    @media ${theme.media.lteSmall} {
      font-size: ${theme.font.sizes.large};
    }
  `,
};

const TitleSize = {
  xxsmall: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xxsmall};
  `,
  xsmall: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xsmall};
  `,
  small: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.small};
  `,
  medium: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.medium};
  `,
  large: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.large};
    ${MediaFont.large(theme)};
  `,
  xlarge: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xlarge};
  `,
  xxlarge: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xxlarge};
    ${MediaFont.xxlarge(theme)};
  `,
  huge: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.huge};
  `,
  xhuge: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xhuge};
  `,
};

const TitleCase = (uppercase: boolean) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

const TitleColor = (
  color:
    | 'white'
    | 'lightGrey1'
    | 'lightGrey2'
    | 'grey'
    | 'darkGrey'
    | 'black'
    | 'blue',
  theme: DefaultTheme,
) => css`
  color: ${color ? theme.colors[color] : theme.colors.black};
`;

export const Container = styled.h1<ContainerProps>`
  ${({ theme, size, color, uppercase }) => css`
    font-family: ${theme.font.family.default};
    font-style: normal;
    font-weight: 700;
    line-height: 1.2;
    text-transform: capitalize;

    ${TitleSize[size](theme)}
    ${TitleCase(uppercase)}
    ${TitleColor(color, theme)}
  `}
`;
