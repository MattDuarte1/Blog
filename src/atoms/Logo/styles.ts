import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.font.family.default};
    font-style: normal;
    font-weight: 600;
    font-size: ${theme.font.sizes.large};
    line-height: 45px;
    position: relative;
    &::first-letter {
      color: ${theme.colors.blue};
    }

    &::after {
      content: '.blog';
      font-family: ${theme.font.family.default};
      font-style: normal;
      font-weight: 600;
      font-size: ${theme.font.sizes.small};
      color: ${theme.colors.blue};
    }
  `}
`;
