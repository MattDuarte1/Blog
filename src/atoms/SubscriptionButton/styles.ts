import styled, { css } from 'styled-components';

export const Container = styled.button`
  ${({ theme }) => css`
    background: ${theme.colors.blue};
    border: none;
    border-radius: 25px;
    display: flex;
    justify-content: center;
    align-items: center;
    gap: 0.9rem;
    padding: 1rem;

    cursor: pointer;
  `}
`;
