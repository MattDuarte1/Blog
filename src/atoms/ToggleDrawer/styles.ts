import styled, { css } from 'styled-components';

type IToggle = {
  active?: boolean;
};

export const Toggle = styled.button<IToggle>`
  ${({ theme, active }) => css`
    background-color: #fff;
    box-shadow: 0 10px 20px rgba(0, 0, 0, 0.08);
    border-radius: 10px;
    border: none;
    overflow: hidden;
    display: none;
    cursor: pointer;
    justify-content: center;
    align-items: center;
    position: relative;
    height: 70px;
    width: 70px;

    p {
      position: absolute;
      width: 40px;
      height: 4px;
      background-color: ${theme.colors.blue};
      border-radius: 4px;
      transition: 0.5s;
    }

    p:nth-child(1) {
      transform: ${active
        ? 'translateY(0px) rotate(45deg)'
        : 'translateY(-15px)'};
      width: ${active ? '40px' : '25px'};
      left: 15px;
    }

    p:nth-child(2) {
      transform: ${active
        ? 'translateY(0px) rotate(315deg)'
        : 'translateY(15px)'};
      width: ${active ? '40px' : '15px'};
      left: 15px;
    }

    p:nth-child(3) {
      transform: ${active && 'translateX(60px)'};
    }

    @media ${theme.media.lteLarge} {
      display: flex;
    }
  `}
`;
