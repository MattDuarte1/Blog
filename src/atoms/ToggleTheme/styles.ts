import styled, { css } from 'styled-components';

export const Container = styled.div`
  position: fixed;
  bottom: 5rem;
  right: 0rem;
  z-index: 2;
`;

export const Label = styled.label`
  position: relative;
  display: inline-block;
  width: 8.9rem;
  height: 4.5rem;
  line-height: 0;
  font-size: 0;
  overflow: hidden;
  color: rgba(0, 0, 0, 0);
`;

export const Input = styled.input`
  ${({ theme }) => css`
    appearance: none;
    opacity: 0;
    width: 0;
    height: 0;
    &:checked + ${Slider} {
      background: ${theme.colors.white};
    }
    &:focus + ${Slider} {
      box-shadow: 0 0 1px ${theme.colors.blue};
    }
    &:checked + ${Slider}:before {
      transform: translateX(4rem);
      background: ${theme.colors.blue};
    }
  `}
`;

export const Slider = styled.span`
  ${({ theme }) => css`
    position: absolute;
    cursor: pointer;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background: ${theme.colors.lightGrey2};
    transition: all 300ms ease-in-out;
    border-radius: 2rem;
    box-shadow: 0 0 50px ${theme.colors.lightGrey2};
    &:before {
      content: '';
      position: absolute;
      height: 4.1rem;
      width: 4.1rem;
      left: 0.4rem;
      bottom: 0.2rem;
      background: ${theme.colors.blue};
      border-radius: 50%;
      transition: all 300ms ease-in-out;
    }
  `}
`;
