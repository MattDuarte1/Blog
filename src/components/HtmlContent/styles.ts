import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    background-color: ${theme.colors.white};
    font-size: calc(${theme.font.sizes.small} + 0.2rem);
    padding: 10rem;
    max-width: 100%;
    line-height: 1.5;

    iframe {
      width: 500px;
    }

    button {
      background-color: ${theme.colors.lightGrey2};
      padding: ${theme.spacings.xsmall};
      cursor: pointer;
      border: none;
    }

    p {
      margin: ${theme.spacings.medium} 0;
    }
    a,
    a:visited,
    a:link {
      color: ${theme.colors.grey};
      text-decoration: none;
      transition: all 300ms ease-in-out;
    }
    a:hover {
      filter: brightness(50%);
    }
    code {
      font-family: monospace;
      color: ${theme.colors.black};
      font-size: ${theme.font.sizes.xsmall};
      background: ${theme.colors.darkGrey};
      padding: 0.2rem;
      margin: 0.2rem;
    }
    pre {
      background: ${theme.colors.darkGrey};
      padding: ${theme.spacings.medium};
      font-family: monospace;
      color: ${theme.colors.black};
      margin: ${theme.spacings.medium} 0;
      width: 100%;
      overflow-x: auto;
      font-size: ${theme.font.sizes.small};
    }
    pre code {
      color: inherit;
      background: inherit;
    }
    img {
      max-width: 100%;
      height: auto;
    }
    hr {
      border: none;
      border-bottom: 0.1rem solid ${theme.colors.lightGrey1};
    }
    ul,
    ol {
      margin: ${theme.spacings.medium} ${theme.spacings.xlarge};
    }
    .table {
      width: 100%;
      overflow: hidden;
      overflow-x: auto;
    }
    table {
      width: 90%;
      border-collapse: collapse;
      margin: ${theme.spacings.medium} 0;
      background-color: ${theme.colors.darkGrey};
    }
    table td,
    table th {
      padding: ${theme.spacings.xsmall};
      border: 0.1rem solid ${theme.colors.lightGrey1};
    }

    blockquote {
      border-left: 0.5rem solid ${theme.colors.lightGrey1};
      color: ${theme.colors.darkGrey};
      filter: brightness(80%);
      padding-left: ${theme.spacings.medium};
      font-style: italic;
      margin: ${theme.spacings.medium};
    }
    @media ${theme.media.lteMedium} {
      font-size: 2rem;
      padding: 3rem;

      iframe {
        width: 100%;
      }
    }
  `}
`;
