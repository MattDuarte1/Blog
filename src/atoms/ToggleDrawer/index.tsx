import * as Styled from './styles';
import { forwardRef } from 'react';

type ToggleDrawerProps = {
  drawerIsOpen: boolean;
  activeDrawer: () => void;
};

const ToggleDrawer = forwardRef<HTMLButtonElement, ToggleDrawerProps>(
  ({ drawerIsOpen, activeDrawer, ...props }, ref) => {
    return (
      <Styled.Toggle
        ref={ref}
        aria-label="menu-toggle"
        data-active={drawerIsOpen}
        active={drawerIsOpen}
        onClick={() => activeDrawer()}
        {...props}
      >
        <p></p>
        <p></p>
        <p></p>
      </Styled.Toggle>
    );
  },
);

ToggleDrawer.displayName = 'Toggle Drawer Component';

export default ToggleDrawer;
