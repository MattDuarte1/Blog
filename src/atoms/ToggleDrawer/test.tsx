import { fireEvent, screen } from '@testing-library/react';
import ToggleDrawer from '.';
import { theme } from '@/styles/theme';
import { renderTheme } from '@/styles/render-theme';

describe('<ToggleDrawer>', () => {
  const activeDrawer = jest.fn();

  it('Should render a ToggleDrawer not actived', () => {
    renderTheme(
      <ToggleDrawer drawerIsOpen={false} activeDrawer={activeDrawer} />,
    );
    const toggleMobile = screen.getByLabelText('menu-toggle');

    expect(toggleMobile).toBeInTheDocument();
    expect(toggleMobile).toHaveAttribute('data-active', 'false');

    expect(toggleMobile.querySelector('p')).toHaveStyle(`width: 25px`);

    expect(toggleMobile).toHaveStyle({
      display: 'none',
    });
  });

  it('Should render a ToggleDrawer actived', () => {
    renderTheme(
      <ToggleDrawer drawerIsOpen={true} activeDrawer={activeDrawer} />,
    );
    const toggleMobile = screen.getByLabelText('menu-toggle');

    expect(toggleMobile).toBeInTheDocument();

    expect(toggleMobile.getAttribute('data-active')).toBe('true');
    expect(toggleMobile.querySelector('p')).toHaveStyle(`width: 40px`);
  });

  it('Should render toggle Mobile when breakpoints screen lteLarge and navbar with display none ', () => {
    renderTheme(
      <ToggleDrawer drawerIsOpen={false} activeDrawer={activeDrawer} />,
    );
    const toggleMobile = screen.getByLabelText('menu-toggle');

    expect(toggleMobile).toHaveStyleRule('display', 'flex', {
      media: theme.media.lteLarge,
    });
  });

  it('Should call activerDrawer function when user clicks toggle button', () => {
    renderTheme(
      <ToggleDrawer drawerIsOpen={false} activeDrawer={activeDrawer} />,
    );
    const toggleMobile = screen.getByLabelText('menu-toggle');

    fireEvent.click(toggleMobile);
    expect(activeDrawer).toHaveBeenCalled();
  });

  it('Should match snapshot toggleDrawer not active', () => {
    renderTheme(
      <ToggleDrawer drawerIsOpen={false} activeDrawer={activeDrawer} />,
    );
    const toggleMobile = screen.getByLabelText('menu-toggle');

    expect(toggleMobile).toMatchSnapshot();
  });

  it('Should match snapshot toggleDrawer actived', () => {
    renderTheme(
      <ToggleDrawer drawerIsOpen={true} activeDrawer={activeDrawer} />,
    );
    const toggleMobile = screen.getByLabelText('menu-toggle');

    expect(toggleMobile).toMatchSnapshot();
  });
});
