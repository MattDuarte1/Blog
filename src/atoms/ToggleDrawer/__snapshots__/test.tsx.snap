// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleDrawer> Should match snapshot toggleDrawer actived 1`] = `
.c0 {
  background-color: #fff;
  box-shadow: 0 10px 20px rgba(0,0,0,0.08);
  border-radius: 10px;
  border: none;
  overflow: hidden;
  display: none;
  cursor: pointer;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  height: 70px;
  width: 70px;
}

.c0 p {
  position: absolute;
  width: 40px;
  height: 4px;
  background-color: #6246EA;
  border-radius: 4px;
  -webkit-transition: 0.5s;
  transition: 0.5s;
}

.c0 p:nth-child(1) {
  -webkit-transform: translateY(0px) rotate(45deg);
  -ms-transform: translateY(0px) rotate(45deg);
  transform: translateY(0px) rotate(45deg);
  width: 40px;
  left: 15px;
}

.c0 p:nth-child(2) {
  -webkit-transform: translateY(0px) rotate(315deg);
  -ms-transform: translateY(0px) rotate(315deg);
  transform: translateY(0px) rotate(315deg);
  width: 40px;
  left: 15px;
}

.c0 p:nth-child(3) {
  -webkit-transform: translateX(60px);
  -ms-transform: translateX(60px);
  transform: translateX(60px);
}

@media (max-width:999px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

<button
  aria-label="menu-toggle"
  class="c0"
  data-active="true"
>
  <p />
  <p />
  <p />
</button>
`;

exports[`<ToggleDrawer> Should match snapshot toggleDrawer not active 1`] = `
.c0 {
  background-color: #fff;
  box-shadow: 0 10px 20px rgba(0,0,0,0.08);
  border-radius: 10px;
  border: none;
  overflow: hidden;
  display: none;
  cursor: pointer;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  height: 70px;
  width: 70px;
}

.c0 p {
  position: absolute;
  width: 40px;
  height: 4px;
  background-color: #6246EA;
  border-radius: 4px;
  -webkit-transition: 0.5s;
  transition: 0.5s;
}

.c0 p:nth-child(1) {
  -webkit-transform: translateY(-15px);
  -ms-transform: translateY(-15px);
  transform: translateY(-15px);
  width: 25px;
  left: 15px;
}

.c0 p:nth-child(2) {
  -webkit-transform: translateY(15px);
  -ms-transform: translateY(15px);
  transform: translateY(15px);
  width: 15px;
  left: 15px;
}

@media (max-width:999px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

<button
  aria-label="menu-toggle"
  class="c0"
  data-active="false"
>
  <p />
  <p />
  <p />
</button>
`;
